From d2eb01e45ccf5db7086ba3a61b03a3cfddd9414a Mon Sep 17 00:00:00 2001
From: Szehon Ho <szehon@cloudera.com>
Date: Thu, 24 Mar 2016 11:12:08 -0700
Subject: [PATCH 0543/1164] CDH-38514 : HIVE-12616 : NullPointerException when
 spark session is reused to run a mapjoin (Nemon
 Lou, via Szehon)

Conflicts:
	ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java

Change-Id: If442bdc7f134a0a3ab08378f74512094ea645779
---
 .../hive/ql/exec/spark/HiveSparkClientFactory.java |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java b/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java
index 274dcf0..99f06e7 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java
@@ -29,6 +29,7 @@
 import org.apache.commons.compress.utils.CharsetNames;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.hive.ql.session.SessionState;
 import org.apache.hadoop.hbase.HBaseConfiguration;
 import org.apache.hadoop.hive.conf.HiveConf;
 import org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch;
@@ -118,6 +119,9 @@ public static HiveSparkClient createHiveSparkClient(HiveConf hiveconf) throws Ex
       sparkMaster = sparkConf.get("spark.master");
       hiveConf.set("spark.master", sparkMaster);
     }
+    if (SessionState.get() != null && SessionState.get().getConf() != null) {
+      SessionState.get().getConf().set("spark.master", sparkMaster);
+    }
     if (sparkMaster.equals("yarn-cluster")) {
       sparkConf.put("spark.yarn.maxAppAttempts", "1");
     }
-- 
1.7.9.5

